pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -372,6 +372,10 @@ func (c *Controller) Cleanup() error {
 }
 
 func (c *Controller) onServiceEvent(pre, curr *v1.Service, event model.Event) error {
+	if !features.EnableExternalNameService && curr.Spec.Type == v1.ServiceTypeExternalName {
```

Comment:
I think this may not work if a service moves to/from ExternalName, we may have stale data?

Reply:
Is that allowed? let me double check

Reply:
fixed